test/http_server_test.exs

```diff
@@ -33,50 +28,26 @@ defmodule HTTPServerTest do
           "Host" => "127.0.0.1 4000",
           "User-Agent" => "ExampleBrowser/1.0"
         },
-        body: "some body"
+        body: ""
       }
 
-      assert HTTPServer.parse_request(message) == expected_parsed_request
+      assert Request.parse_request(message) == expected_parsed_request
```

Comment:
something I noticed about these "`http_server_test`" tests, they are actually testing the `HTTPServer.Request.parse_request/1` function.

Since `HTTPServer` is "main", you don't really need to test "main"

I think this file should be moved to reflect what its actually unit testing, `HTTPServer.Request` aka `request_spec.ex`

Reply:
I wonder what other unit tests could be added as well? (example: `response_spec.exs`)

Reply:
I like this idea! But I might fold them into the Simple HEAD ticket since this PR is pretty big.